robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowGestureDetector.java

```diff
@@ -15,33 +19,32 @@
   @RealObject
   private GestureDetector realObject;
 
-  private MotionEvent onTouchEventMotionEvent;
-  private boolean onTouchEventNextReturnValue = true;
+  private static GestureDetector lastActiveGestureDetector;
 
+  private MotionEvent onTouchEventMotionEvent;
   private GestureDetector.OnGestureListener listener;
-  private static GestureDetector lastActiveGestureDetector;
   private OnDoubleTapListener onDoubleTapListener;
 
-  public void __constructor__(GestureDetector.OnGestureListener listener) {
-    __constructor__(null, listener);
-  }
-
-  public void __constructor__(Context context, GestureDetector.OnGestureListener listener) {
+  public void __constructor__(Context context, GestureDetector.OnGestureListener listener, Handler handler) {
+    Shadow.invokeConstructor(GestureDetector.class, realObject,
+        from(Context.class, context),
+        from(GestureDetector.OnGestureListener.class, listener),
+        from(Handler.class, handler));
     this.listener = listener;
-    if (listener instanceof OnDoubleTapListener) {
-      setOnDoubleTapListener((OnDoubleTapListener) listener);
-    }
   }
 
   @Implementation
   public boolean onTouchEvent(MotionEvent ev) {
     lastActiveGestureDetector = realObject;
     onTouchEventMotionEvent = ev;
-    return onTouchEventNextReturnValue;
+
+    return directlyOn(realObject, GestureDetector.class, "onTouchEvent", from(MotionEvent.class, ev));
```

Comment:
Can you use the directlyOn() version that returns an object instead, slightly better performance but also type safe?